### PR TITLE
remove BaseCollection from the global scope and add a light .jshintrc file

### LINKED ISSUE #424: BaseCollection is in the global scope
`BaseCollection` defined in `shared/base/collection.js` is in the global scope.

### PR DESCRIPTION
**Pull request by ymainier**
Fix #424 
Add a light .jshintrc file and .jshintignore file.

### CODE REVIEW

File: shared/base/collection.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ if (!isServer) {
   Backbone.$ = window.$ || require('jquery');
 }
 
-BaseCollection = Super.extend({
+var BaseCollection = Super.extend({
```
Comment (saponifi3d) [inline shared/base/collection.js]: well that isn't good. thanks for finding that!

File: shared/syncer.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ if (isServer) {
   var serverOnly_qs = 'qs';
   var qs = require(serverOnly_qs);
 } else {
-  Backbone.$ = window.$ || require('jquery');
+  var $ = window.$ || require('jquery');
```
Comment (saponifi3d) [inline shared/syncer.js]: I don't think this is necessary, can you explain why you decided to change this?

File: .jshintrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (saponifi3d) [inline .jshintrc]: I've been thinking about adding a jshint for a bit.. @spikebrehm did you have any thoughts against jshint?

**Comment by saponifi3d** [comment]
Overall, this is awesome. i just had the one question about why you split up the assignment for `Backbone.$`  Thanks for the fix!

File: shared/syncer.js (hunk 2 above)
Comment (ymainier) [inline shared/syncer.js]: jshint was complaining about $ not being on line 138. (`$.getJSON(url, this.params, function(resp) {"`)
I sax two options to fix this : the one above or using `window.$.getJSON`

Which one do you prefer ?

Comment (saponifi3d) [inline shared/syncer.js]: ah, if it's for jshint then :+1: wasn't sure if you had any other reasons for doing it.

File: .jshintrc (hunk 3 above)
Comment (spikebrehm) [inline .jshintrc]: sounds like a good idea to me

File: shared/base/collection.js (hunk 1 above)
Comment (spikebrehm) [inline shared/base/collection.js]: :fearful: that's my error in converting from CoffeeScript

**Pull request merged**